lib/ansible/modules/windows/win_netbios.ps1

```diff
@@ -0,0 +1,65 @@
+#!powershell
+
+# Copyright: (c) 2019, Thomas Moore (@tmmruk) <hi@tmmr.uk>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+    options = @{
+        state = @{ type = "str"; choices = "enabled", "disabled", "default"; required = $true }
+        adapter_names = @{ type = "list"; required = $false }
+    }
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$state = $module.Params.state
+$adapter_names = $module.Params.adapter_names
+
+switch ( $state )
+{
+    'default'{ $netbiosoption = 0 }
+    enabled { $netbiosoption = 1 }
+    disabled { $netbiosoption = 2 }
+}
+
+if(-not $adapter_names)
+{
+    # Target all network adapters on the system
+    $target_adapters_config = Get-CimInstance -Class Win32_NetworkAdapterConfiguration | Where-Object {$_.IPEnabled -eq $true}
+}
+else
+{
+    $target_adapters = Get-CimInstance -Class Win32_NetworkAdapter | Where-Object NetConnectionId -in $adapter_names | Select-Object MacAddress,NetConnectionId
```

Comment:
Same thing applies here, you should be using the `-Filter` and `-Property` fields instead. Having the filter check an array makes things a bit more difficult but you can still do it. You are also better off leaving WMI to do the association to get the Win32_NetworkAdapterConfiguration like so;

```
$get_params = @{
    Class = 'Win32_NetworkAdapter'
    Filter = ($adapter_names | ForEach-Object -Process { "NetConnectionId='$_'" }) -join " OR "
    KeyOnly = $true
}
$target_adapters_config = Get-CimInstance @get_params | Get-CimAssociatedInstance -ResultClass 'Win32_NetworkAdapterConfiguration'
```

This works because Win32_NetworkAdapterConfiguration is an associated class of Win32_NetworkAdapter and WMI will automatically link the 2 together for you.

Reply:
Again thanks for the tip. Have added this into the code now.